packages/valory/skills/abstract_round_abci/base.py

```diff
@@ -1858,16 +1858,29 @@ def __init__(
         self._timeouts = Timeouts[EventType]()
         self._reset_index = 0
         self._is_termination_set = (
-                self.termination_round_cls is not None
-                and self.termination_transition_function is not None
-                and self.termination_event is not None
+            self.background_round_cls is not None
+            and self.termination_transition_function is not None
+            and self.termination_event is not None
         )
 
     @classmethod
     def is_abstract(cls) -> bool:
         """Return if the abci app is abstract."""
         return cls._is_abstract
 
+    @classmethod
+    def add_termination(
+        cls,
+        background_round_cls: Type[AppState],
+        termination_event: EventType,
+        termination_abci_app: "Type[AbciApp]",
+    ) -> "Type[AbciApp]":
+        """Sets the termination related class variables."""
+        cls.background_round_cls = background_round_cls
+        cls.termination_transition_function = termination_abci_app.transition_function
+        cls.termination_event = termination_event
+        return cls
+
```

Comment:
The use would be:
```python
AnyAbciApp = chain(
    (
        RegistrationAbci.AgentRegistrationAbciApp,
        SafeDeploymentAbci.SafeDeploymentAbciApp,
        TransactionSubmissionAbci.TransactionSubmissionAbciApp,
        ResetAndPauseAbci.ResetPauseABCIApp,
    ),
    abci_app_transition_mapping,
).add_termination(BackgroundRound, Event.TERMINATE, TerminationAbciApp)
```
or
```python
AnyAbciApp.add_termination(BackgroundRound, Event.TERMINATE, TerminationAbciApp)
```

Reply:
Because the arguments here will always be the same, we can have a `with_termination()` function in `termination_abci` that would automatically add these. 
The use of it could be something like:
```python
with_termination(AnyAbciApp)
```

Reply:
I like that it can be used as an optional extension. Are there specific reasons / advantages to doing this over having it within the `chain` call?

Reply:
Yes the separation is desirable.